File: MetroRappid/CAPRouteMapViewController.m
//
//  CAPRouteMapViewController.m
//  MetroRappid
//
//  Created by Luq on 5/26/14.
//  Copyright (c) 2014 Createch. All rights reserved.
//

#import "CAPRouteMapViewController.h"
#import "GTFSDB.h"  // FIXME: Remove whatever calls this into a dataSource-esque class
#import "CAPTrip.h"


@interface CAPRouteMapViewController ()

@end


@implementation CAPRouteMapViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.navigationItem.title = self.route.routeId;
    
    NSMutableArray *tripsData = [GTFSDB activeTripsForRoute:self.route.routeId];
    NSMutableArray *trips = [[NSMutableArray alloc] init];
    for (NSDictionary *tripData in tripsData) {
        CAPTrip *trip = [[CAPTrip alloc] init];
        [trip updateWithGTFS:tripData];
        [trips addObject:trip];
    }
    NSLog(@"Got %d trips", (int)trips.count);

    NSMutableArray *shapes = [GTFSDB activeTripsForRoute:self.route.routeId];
    
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}

@end

File: MetroRappid/CAPRouteMapViewController.swift
//
//  CAPRouteMapViewController.swift
//  MetroRappid
//
//  Created by Luq on 6/3/14.
//  Copyright (c) 2014 Createch. All rights reserved.
//

import Foundation
import UIKit
import MapKit


class CAPRouteMapViewController: UIViewController, MKMapViewDelegate {
    
    var route: CAPRoute!
    var trip0: CAPTrip
    var trip1: CAPTrip
    
    @IBOutlet var mapView : MKMapView
    @IBOutlet var tripControl : UISegmentedControl
    
    init(coder aDecoder: NSCoder!)
    {
        trip0 = CAPTrip()
        trip1 = CAPTrip()
        super.init(coder: aDecoder)
    }
    
    init(nibName nibNameOrNil: String!, bundle nibBundleOrNil: NSBundle!) {
        trip0 = CAPTrip()
        trip1 = CAPTrip()
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
    }
    
    override func viewDidLoad() {
        mapView.delegate = self
        
        println("did load \(self)")
        
        var tripsData: Array = GTFSDB.tripsForRoute(self.route.routeId)
        
        trip0.updateWithGTFS(tripsData[0] as NSMutableDictionary)
        trip1.updateWithGTFS(tripsData[1] as NSMutableDictionary)
        
        tripControl.removeAllSegments()
        tripControl.insertSegmentWithTitle(trip0.tripHeadsign, atIndex: 0, animated: false)
        tripControl.insertSegmentWithTitle(trip1.tripHeadsign, atIndex: 1, animated: false)
        tripControl.selectedSegmentIndex = 0
    }
    
    
    
}